_posts/2022-12-07-jberet-quarkus.md

```diff
@@ -0,0 +1,123 @@
+---
+layout:     post
+title:      "Using JBeret With Quarkus"
+subtitle:   ""
+date:       2022-12-07
+author:     Wei Nan Li
+---
+
+JBeret can be used within Quarkus:
+
+- [GitHub - quarkusio/quarkus: Quarkus: Supersonic Subatomic Java.](https://github.com/quarkusio/quarkus)
+
+In this article I won’t go into details of Quarkus. If you are not familiar with Quarkus, you can learn and check the above project link. Quarkus has the JBeret extension:
+
+- [Quarkus JBeret Extension](https://github.com/quarkiverse/quarkus-jberet)
+
+Here is the class diagram of the core part of `quarkus-jberet`:
+
+![](https://raw.githubusercontent.com/jberet/jberet.github.io/main/_imgs/jberet_quarkus/class.jpg)
+
+From the above class diagram, we can see it adds a layer for JBeret to be enabled in Quarkus, so the project that uses JBeret can be deployed into Quarkus with minimum configuration. We can clone this project and play with the tests inside the project:
+
+```bash
+➤ git clone https://github.com/quarkiverse/quarkus-jberet.git
+```
+
+There are multiple tests that can be used as example:
+
+- [quarkus-jberet/core/deployment/src/test/java/io/quarkiverse/jberet/deployment at main · quarkiverse/quarkus-jberet · GitHub](https://github.com/quarkiverse/quarkus-jberet/tree/main/core/deployment/src/test/java/io/quarkiverse/jberet/deployment)
+
+One of the above tests is `BatchletTest`:
+
+- https://github.com/quarkiverse/quarkus-jberet/blob/main/core/deployment/src/test/java/io/quarkiverse/jberet/deployment/BatchletTest.java
+
+This test uses the Quarkus test framework to set up the service running environment, and it will start the Quarkus before test, and then register the components and run the test. Here is the setup part of the test:
+
+```java
+@RegisterExtension
+static QuarkusUnitTest TEST = new QuarkusUnitTest()
+    .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+        .addClasses(DummyBatchlet.class)
+        .addAsManifestResource("batchlet.xml", "batch-jobs/batchlet.xml"));
+```
+
+In the above code, it creates the test JAR archive, and it puts the `batchlet.xml` into the JAR file. This `batchlet.xml` is defined in the test code directory:
+
+- https://github.com/quarkiverse/quarkus-jberet/blob/main/core/deployment/src/test/resources/batchlet.xml
+
+The `batchlet.xml` defines a basic `batchlet` for testing:
+
+```xml
+<job id="batchlet-job" xmlns="http://xmlns.jcp.org/xml/ns/javaee" version="1.0">
```

Comment:
the batch job xml is named `batchlet.xml`, but the job id is `batchlet-job`, which is confusing. We should keep them consistent by renaming the file to `batchlet-job.xml`.

Reply:
The example is directly quoted from Quarkus JBeret project so I can't directly modify it in the blog post. I modified the text to address this.